### ISSUE TITLE
Prototype assignment (replacement)

### ISSUE THREAD

**Issue by KACAH** [MikeMcl/decimal.js#161]
Hey!

Right now [Decimal prototype](https://github.com/MikeMcl/decimal.js/blob/be7d6b05ab534cf1eda0a1d0a2558e8f049583fe/decimal.js#L4356) is replaced by completely new [object P](https://github.com/MikeMcl/decimal.js/blob/be7d6b05ab534cf1eda0a1d0a2558e8f049583fe/decimal.js#L125). According to [MDN article](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Inheritance_and_the_prototype_chain) about inheritance and prototypes this is considered a bad practice:

> do not set the prototype f.prototype = {b:3,c:4}; this will break the prototype chain

The consequence is - some libraries can't identitfy Decimal instances as custom JS objects and break them trying to clone, serialize, etc. The example of such library is MobX v6. which is checking "plain" objects [like this](https://github.com/mobxjs/mobx/blob/8ae18a91d554f2d000f2f617d69204a8cd30ed07/src/utils/utils.ts#L87).

```javascript
const plainObjectString = Object.toString()

export function isPlainObject(value) {
    if (!isObject(value)) return false
    const proto = Object.getPrototypeOf(value)
    if (proto == null) return true
    return proto.constructor?.toString() === plainObjectString
}
```

The result for Decimal instances is 'true', which is incorrect.

Is it possible to replace prototype assignment with prototype **properties** assignment? 

For instance replace

```javascript
Decimal.prototype = P;
```

with

```javascript
for (var protoKey in P) {
  if (Object.prototype.hasOwnProperty.call(P, protoKey)) {
    Decimal.prototype[protoKey] = P[protoKey];
  }
}
```

**Comment by MikeMcl** [comment]
Here the constructor property is set on the instance not the prototype because the prototype may be shared between multiple Decimal constructors, so the value of the constructor property may be different for different instances.

The issue with the `isPlainObject` function returning `true` could be solved by adding the following after the shared prototype object is assigned to `Decimal.prototype`.

```javascript
Decimal.prototype.constructor = Decimal;
```

I have not included this previously because `Decimal.prototype.constructor` is shadowed by `instance.constructor`, but yes, it would solve this issue so I'll consider adding it to the next release.

Thanks for your input.

**Comment by KACAH** [comment]
Thank you!

**Comment by MikeMcl** [comment]
Fixed in *v10.3.0*.
